### PR TITLE
Refreshing the MA module to use latest swagger

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Comment by anuchandy** [comment]
@samvaity, I tried to keep the convenience layer changes minimum, i.e. changes to get it compiled with breaking changes that swagger introduced. Once the protocol layer updates are in, new features need to be added, which includes the following:

- Add LogAnalytics, AzureEventHubs data feed source API.
- Expose DataFeedId in DataPointAnomaly and AnomalyIncident models.
- Add splitAlertByDimensions (string[]) to AnomalyAlertConfiguration.
- Add value and expected value to Anomaly and incident Result.
- Expose type in BoundaryCondition model types.
- Finalize the shape of the new credential API and its implementation.

This pr includes the following feature changes: (which was needed to get code compilable and testable).

- Expose Alert and detection config listing as pageable APIs.
- Remove Elasticsearch and HttpRequest data sources APIs.
- Adapt new models SuppressConditionPatch, SmartDetectionConditionPatch, HardThreasholdConditionPatch, ChangeThreasholdConditionPatch, WholeMetricConfigurationPatch in the SDK implementation.

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/MetricsAdvisorAdministrationAsyncClient.java [hunk 1]
```diff
@@ -1585,9 +1591,10 @@
      * Fetch the anomaly alert configurations associated with a detection configuration.
      *
      * <p><strong>Code sample</strong></p>
-     * {@codesnippet com.azure.ai.metricsadvisor.administration.MetricsAdvisorAdministrationAsyncClient.listAnomalyAlertConfigs#String}
+     * {@codesnippet com.azure.ai.metricsadvisor.administration.MetricsAdvisorAdministrationAsyncClient.listAnomalyAlertConfigs#String-ListAnomalyAlertConfigsOptions}
      *
      * @param detectionConfigurationId The id of the detection configuration.
+     * @param options th e additional configurable options to specify when querying the result.
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/administration/MetricsAdvisorAdministrationAsyncClient.java:1597]: nit:space between `th e`

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/AnomalyIncident.java [hunk 2]
```diff
@@ -102,6 +129,25 @@ public DimensionKey getRootDimensionKey() {
         return this.rootDimensionKey;
     }
 
+    /**
+     * Gets the aggregated value at root dimension node where the incident anomalies
+     * are logically aggregated.
+     *
+     * @return The value.
+     */
+    public Double getValue() {
```
Comment (samvaity) [inline sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/AnomalyIncident.java:138]: Should dthis be primitive or can be nullable?

**Review by samvaity — APPROVED** [review]
Changes lgtm :)

File: sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/AnomalyIncident.java (hunk 2 above)
Comment (anuchandy) [reply sdk/metricsadvisor/azure-ai-metricsadvisor/src/main/java/com/azure/ai/metricsadvisor/models/AnomalyIncident.java:138]: It should be primitive (let me confirm), let's address that in a follow-up pr just to unblock us from adding the rest of the features.

**Pull request merged**